File: Python3.x/307-Range_Sum_Query_-_Mutable.py
# Runtime: 748 ms, faster than 6.47% of Python3 online submissions for Range Sum Query - Mutable.
# Memory Usage: 16.6 MB, less than 89.58% of Python3 online submissions for Range Sum Query - Mutable.
class NumArray:

    def __init__(self, nums: List[int]):
        self.nums = nums

    def update(self, i: int, val: int) -> None:
        self.nums[i] = val

    def sumRange(self, i: int, j: int) -> int:
        return sum(self.nums[i:j + 1])


# Your NumArray object will be instantiated and called as such:
# obj = NumArray(nums)
# obj.update(i,val)
# param_2 = obj.sumRange(i,j)

File: Algorithms/C++/307-Range_Sum_Query_-_Mutable.cpp
// Runtime: 144 ms, faster than 24.61% of C++ online submissions for Range Sum Query - Mutable.
// Memory Usage: 18.8 MB, less than 99.23% of C++ online submissions for Range Sum Query - Mutable.
class NumArray {
public:
    NumArray(vector<int>& nums) {
        vec.swap(nums);
    }
    
    void update(int i, int val) {
        vec[i] = val;
    }
    
    int sumRange(int i, int j) {
        int res = 0;
        for (; i <= j; ++i)
            res += vec[i];
        return res;
    }
private:
    vector<int> vec;
};

/**
 * Your NumArray object will be instantiated and called as such:
 * NumArray* obj = new NumArray(nums);
 * obj->update(i,val);
 * int param_2 = obj->sumRange(i,j);
 */